components/Table/Table.js

```diff
@@ -95,21 +152,139 @@ export class Table {
+    });
+
+    btnClean.addEventListener('click', () => {
+      filterValue.value = '';
+      document
+        .querySelectorAll('.table-row')
+        .forEach((row) => (row.style.display = 'flex'));
+    });
+
+    const filterFunction = (e) => {
+      let filter = e.target.value;
+      this.callbacksMap.filterCallback(filter);
+    };
+
+    filterValue.addEventListener('input', filterFunction);
+
+    filterMenuContainer.append(
+      btnClean,
+      filterLabel,
+      filterOperator,
+      filterValue,
+      btnExit
+    );
+    this.tableHeader.append(filterMenuContainer);
+  }
+
+  createPagination(TOTAL_AMOUNT = 0) {
+    const pagintaionContainer = document.createElement('div');
+    pagintaionContainer.classList.add('pagination-container');
+
+    const dropdownPagintaionContainer = document.createElement('select');
+    dropdownPagintaionContainer.classList.add('dropdown-pagination-container');
+
+    const pageNumberContainer = document.createElement('ul');
+    pageNumberContainer.classList.add('page-num-container');
+
+    const renderPagesNumbers = () => {
+      const pages = Math.ceil(TOTAL_AMOUNT / this.amount);
+
+      for (let i = 1; i <= pages; i++) {
+        const pageNumberBtn = document.createElement('li');
+        pageNumberBtn.classList.add('page-number');
+        pageNumberBtn.textContent = i;
+        pageNumberContainer.append(pageNumberBtn);
+
+        pageNumberBtn.addEventListener('click', () => {
+          const currentPage = +pageNumberBtn.innerHTML;
+          this.callbacksMap.paginationCallback(currentPage, this.amount);
+        });
+      }
+    };
+    renderPagesNumbers();
+
+    const dropdownPagintaionItems = this.PaginationConfigList.map((item) => {
+      const dropPagItem = document.createElement('option');
+      dropPagItem.classList.add('drop-pag-item');
+      dropPagItem.textContent = item.pageSize;
+
+      dropdownPagintaionContainer.addEventListener('change', (e) => {
+        if (this.amount !== +e.target.value) {
+          pageNumberContainer.innerHTML = '';
+          this.amount = +e.target.value;
+          renderPagesNumbers();
+          this.callbacksMap.sortCallback();
+        }
+      });
+
+      return dropPagItem;
+    });
+
+    dropdownPagintaionContainer.append(...dropdownPagintaionItems);
+    pagintaionContainer.append(
+      dropdownPagintaionContainer,
+      pageNumberContainer
+    );
+    this.tableContainer.append(pagintaionContainer);
+  }
+
   clearTable() {
     this.tableBody.innerHTML = '';
   }
 
   render(data) {
     this.clearTable();
     const fragment = new DocumentFragment();
-    data.forEach((element) => {
+    let hidePagination = document.querySelector('.page-num-container');
+    let paginationData = data.slice(0, this.amount);
+    let dataLenght = Object.keys(paginationData).length;
+    if (dataLenght < this.amount) {
+      hidePagination.style.display = 'none';
+    } else {
+      hidePagination.style.display = 'flex';
+    }
```

Comment:
This should not be in the render method
This is not related to the render
This is related to the pagination and you could decide to render the pagination page buttons or not in the separate method, that you should do instead of `renderPagesNumbers` function that you have in the `createPagination` method now